Add support for nested comments

SydorukDmytro opened the pull request:
Add support for nested comments

Introduced hierarchical comment support by adding a `ParentId` property to enable parent-child relationships. Updated the `Comment` entity, `CommentDto`, and `CommentRequest` to include `ParentId` and `Children` properties. Enhanced `CommentService` with a `BuildTree` method to return comments as a tree structure.

Modified `CommentController` to accept `CommentRequest` for creating nested comments. Updated `ICommentService` and `CommentsRepository` to handle parent-child relationships and include user details in comments.

Added a migration to update the `Comments` table with a `ParentId` column and foreign key constraints. Updated `CommentConfiguration` and `DataBaseMappings` to reflect these changes. Improved the data model and API to support nested comment creation and retrieval.

Commit:
Add support for nested comments

Introduced hierarchical comment support by adding a `ParentId` property to enable parent-child relationships. Updated the `Comment` entity, `CommentDto`, and `CommentRequest` to include `ParentId` and `Children` properties. Enhanced `CommentService` with a `BuildTree` method to return comments as a tree structure.

Modified `CommentController` to accept `CommentRequest` for creating nested comments. Updated `ICommentService` and `CommentsRepository` to handle parent-child relationships and include user details in comments.

Added a migration to update the `Comments` table with a `ParentId` column and foreign key constraints. Updated `CommentConfiguration` and `DataBaseMappings` to reflect these changes. Improved the data model and API to support nested comment creation and retrieval.

YuriLukashenko commented on Lumix.Application/Services/CommentService.cs:
```diff
@@ -13,24 +13,49 @@ public CommentService(ICommentsRepository commentsRepository)
+		public async Task<IEnumerable<CommentDto>> GetByPhotoId(Guid photoId)
 		{
-			return await _commentsRepository.GetByPhotoId(photoId);
+			var flatDtos = await _commentsRepository.GetByPhotoId(photoId)
```
чому неймінг flatDtos?

Commit:
refactor(api): switch comment creation to JSON input and return full CommentDto
- Replaced [FromForm] with [FromBody] to support JSON-based requests
- Updated controller to return the created CommentDto instead of void
- Modified service to return fully populated CommentDto after insertion
- Updated repository to load navigation properties and map entity to CommentDto

Commit:
Renamed flatDtos to allComments
